src/gui/plugins/component_inspector/Physics.qml

```diff
@@ -0,0 +1,262 @@
+/*
+ * Copyright (C) 2021 Open Source Robotics Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+*/
+import QtQuick 2.9
+import QtQuick.Controls 1.4
+import QtQuick.Controls 2.2
+import QtQuick.Controls.Material 2.1
+import QtQuick.Dialogs 1.0
+import QtQuick.Layouts 1.3
+import QtQuick.Controls.Styles 1.4
+import "qrc:/ComponentInspector"
+import "qrc:/qml"
+
+// Item displaying physics information.
+Rectangle {
+  height: header.height + content.height
+  width: componentInspector.width
+  color: index % 2 == 0 ? lightGrey : darkGrey
+
+  // Left indentation
+  property int indentation: 10
+
+  // Minimum parameter value, step size and RTF must be strictly positive
+  property double minPhysParam: 0.000001
+
+  // Maximum parameter value
+  property double maxPhysParam: 100000
+
+  // Horizontal margins
+  property int margin: 5
+
+  property int iconWidth: 20
+  property int iconHeight: 20
+
+  // Loaded item for physics step size
+  property var stepSizeItem: {}
+
+  // Loaded item for real time factor
+  property var realTimeFactorItem: {}
+
+  // Send new physics data to C++
+  function sendPhysics() {
+    // TODO(anyone) There's a loss of precision when these values get to C++
+    componentInspector.onPhysics(
+      stepSizeItem.value,
+      realTimeFactorItem.value
+    );
+  }
+
+  FontMetrics {
+    id: fontMetrics
+    font.family: "Roboto"
+  }
+
+  /**
+   * Used to create a spin box
+   */
+
+  Component {
+    id: writablePositiveNumber
+    IgnSpinBox {
+      id: writableSpin
+      value: writableSpin.activeFocus ? writableSpin.value : numberValue
+      minimumValue: minPhysParam
+      maximumValue: maxPhysParam
+      decimals: 6
```

Comment:
I recommend setting a small step size like `stepSize: 0.001`. The default is 1 and it causes things to explode fast :sweat_smile: